tests/test_als_implicit.py

```diff
@@ -41,11 +41,102 @@ def test_als_predict_basic():
     algo.fit(simple_df)
 
     preds = algo.predict_for_user(10, [3])
+    print(preds)
     assert len(preds) == 1
     assert preds.index[0] == 3
     assert preds.loc[3] >= -0.1
     assert preds.loc[3] <= 5
 
+def test_als_predict_basic_for_new_ratings():
+    algo = als.ImplicitMF(20, iterations=10)
+    algo.fit(simple_df)
+
+    new_ratings = pd.Series([4.0, 5.0], index=[1, 2]) # items as index and ratings as values
+
+    preds = algo.predict_for_user(15, [3], new_ratings)
+
+    assert len(preds) == 1
+    assert preds.index[0] == 3
+    assert preds.loc[3] >= -0.1
+    assert preds.loc[3] <= 5
+
+def test_als_predict_basic_for_new_user_with_new_ratings():
+    u = 10
+    i = 3
+
+    algo = als.ImplicitMF(20, iterations=10)
+    algo.fit(simple_df)
+
+    preds = algo.predict_for_user(u, [i])
+
+    new_u_id = 1
+    new_ratings = pd.Series([4.0, 5.0], index=[1, 2]) # items as index and ratings as values
+
+    new_preds = algo.predict_for_user(new_u_id, [i], new_ratings)
+    assert abs(preds.loc[i] - new_preds.loc[i]) <= 0.1
+
+def test_als_predict_for_new_users_with_new_ratings():
+    n_users = 3
```

Comment:
Add a docstring describing in more words (1-2 sentences) the purpose of these tests.